addons/zylann.editor_debugger/dock.gd

```diff
@@ -56,10 +56,11 @@ func get_tree_view() -> Tree:
 
 func _ready() -> void:
 	_popup_menu.clear()
-	for id in _popup_action_names:
-		_popup_menu.add_item(_popup_action_names[id].title, id)
+	for id: int in _popup_action_names:
+		var popup_data: Dictionary = _popup_action_names[id]
+		_popup_menu.add_item(str(popup_data.title), id)
```

Comment:
I'm not very pleased by the fact `str` would have to be used here. Because in principle, if `title` was mistakenly not a string, that mistake would not be spotted anymore because `str` works on everything.

Reply:
Thanks for your comments, I'll update the PR soon.

However about this one in particular: I wasn't sure how else to avoid type safety errors here. The code as before triggered an Unsafe Call Argument error, and with `as String` an Unsafe Cast error.

Reply:
I have since discovered that (very strangely) the use of `str` can be avoided here via `etc as Variant as String`. Note that `etc as String` alone does not work... just `as Variant as String`. I will update it to use this casting arcana instead of `str` in a new commit.

Reply:
>  Unsafe Call Argument error

Also why are these errors at all? GDScript has typing, but isn't a strongly typed. It is optional. It shouldn't be an absolutely vital necessity to forcefully type every corner of code.

Reply:
GDScript's performance does in fact benefit from typing.

https://www.beep.blog/2024-02-14-gdscript-typing/

Though the main reason that I personally keep type errors like these turned on is because, though I do stumble over weird things like this from time to time, they also save me a whole lot of time by catching my typos and other dumb mistakes right away as I'm writing the code, and not just later on as I'm trying to run it.

Reply:
> GDScript's performance does in fact benefit from making use of types.

I know that, but this plugin and that area of it are not in critical needs for performance.

This is a valid reason, but I find it quite annoying that such policy would have to contaminate every plugin like that, it's not even code from your project. While I also like having typed GDScript, limitations of the language makes it really obnoxious and unnecessary to enforce that far.
Is there really no way at all that this area using an inherently untyped thing (Dictionary) can't simply use... legit untyped GDScript?

Reply:
Which warning is `as String` producing when you aren't using `as Variant as String`?

Reply:
> Maybe something like this?

I have no idea why that works and passing it directly doesn't. But it does work. I'll go ahead and make the change.

> While I also like having typed GDScript, limitations of the language makes it really obnoxious and unnecessary to enforce that far.

I agree. I confess I really don't like GDScript, and that's mostly due to the shoddiness of its type system. If there weren't so many shortcomings with C# support then I would be using that without looking back. It's not my favorite language to work with, but it's a hell of a lot better than GDScript.

But alas. I still can't even properly document my node and resource types for the editor when using C#. https://github.com/godotengine/godot-proposals/issues/8269

> This is a valid reason, but I find it quite annoying that such policy would have to contaminate every plugin like that, it's not even code from your project. 

Yep. I agree with this too. But while I'm working on my own EditorPlugin for my project, if I want to both have type safety errors in my own code and also use this (very useful, thank you!) addon, there is no way except to make this addon comply with all the same error settings I have configured for my own code. (Where it really is more of a benefit than it is a pain. Not realizing that I typo'd a function's arguments until runtime is such a pain.)

Maybe this feature to apply GDScript error settings to addons selectively, instead of just all-or-nothing, will get merged for 4.4 and then this won't be a problem anymore. But for now, there's not a better way. https://github.com/godotengine/godot/pull/93889

Reply:
> Which warning is `as String` producing when you aren't using `as Variant as String`? IMO it isn't normal that multiple casts like that would have to be written for something that can't possibly contain anything but Variants already.

![image](https://github.com/user-attachments/assets/9fff874b-6043-45ef-b0e8-c885ba1afed3)